testsuites/benchmark/src/main/java/org/eclipse/rdf4j/benchmark/rio/ParserBenchmark.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2021 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+
+package org.eclipse.rdf4j.benchmark.rio;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import org.eclipse.rdf4j.benchmark.rio.util.BlackHoleRDFHandler;
+import org.eclipse.rdf4j.benchmark.rio.util.DataSetGenerator;
+import org.eclipse.rdf4j.rio.RDFFormat;
+import org.eclipse.rdf4j.rio.RDFParser;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.infra.Blackhole;
+
+/**
+ * Parser Benchmark implementation that consumer a file created through {@link DataSetGenerator} with
+ * {@link BlackHoleRDFHandler}.
+ *
+ * @author Tomas Kovachev t.kovachev1996@gmail.com
+ */
+@State(Scope.Benchmark)
+public abstract class ParserBenchmark {
+	public static final String XMS = "-Xms4g";
+	public static final String XMX = "-Xmx4g";
+	public static final int WARM_UPS = 3;
+	public static final int EXECUTIONS = 3;
+	public static final int TOTAL_STATEMENTS = 1_000_000;
+	public static final int MIN_STRING_LENGTH = 100;
+	public static final int MAX_STRING_LENGTH = 300;
+	public static final int PERCENT_BNODE = 30;
+	public static final int PERCENT_LITERALS = 50;
+	public static final boolean TEXT_ONLY = false;
```

Comment:
Could you use annotations instead? I like to run the benchmarks with the JMHack plugin in IntelliJ instead of with a Runner.

Here is how I do annotations: https://github.com/eclipse/rdf4j/blob/main/core/sail/memory/src/test/java/org/eclipse/rdf4j/sail/memory/benchmark/QueryBenchmark.java